### PR TITLE
normalize emails on bulk invite, normalize/lowercase emails on invite…

### PR DESCRIPTION
**Pull request by rkuo-danswer**
… matching

## Description
Fixes DAN-432.  I did end up normalizing emails when inserting via bulk invite as the email_validator docs strongly suggest to follow this pattern.

## How Has This Been Tested?
Ran bulk invite, ran tests against upper and lowercase signup.

## Accepted Risk
[Any know risks or failure modes to point out to reviewers]

## Related Issue(s)
[If applicable, link to the issue(s) this PR addresses]

## Checklist:
- [ ] All of the automated tests pass
- [ ] All PR comments are addressed and marked resolved
- [ ] If there are migrations, they have been rebased to latest main
- [ ] If there are new dependencies, they are added to the requirements
- [ ] If there are new environment variables, they are added to all of the deployment methods
- [ ] If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- [ ] Docker images build and basic functionalities work
- [ ] Author has done a final read through of the PR right before merge

### CODE REVIEW

File: backend/danswer/server/manage/users.py [hunk 1]
```diff
@@ -159,12 +160,18 @@ def bulk_invite_users(
...
         raise HTTPException(
             status_code=400, detail="Auth is disabled, cannot invite users"
         )
 
-    all_emails = list(set(emails) | set(get_invited_users()))
+    normalized_emails = []
+    for email in emails:
+        email_info = validate_email(email)  # can raise EmailNotValidError
+        normalized_emails.append(email_info.normalized)  # type: ignore
+    all_emails = list(set(normalized_emails) | set(get_invited_users()))
```
Comment (yuhongsun96) [inline backend/danswer/server/manage/users.py:174]: any potential issue with having duplicates (like a normalized and a non normalized version) in the DB?

**Review by yuhongsun96 — APPROVED** [review]

File: backend/danswer/auth/users.py [hunk 2]
```diff
@@ -8,6 +8,8 @@
 from typing import Optional
 from typing import Tuple
 
+from email_validator import EmailNotValidError
```
Comment (Weves) [inline backend/danswer/auth/users.py:11]: is there a reason why we use `from pydantic import validate_email` below but `from email_validator import validate_email` here?

Comment (rkuo-danswer) [reply backend/danswer/auth/users.py:11]: good catch, thanks!

File: backend/danswer/server/manage/users.py (hunk 1 above)
Comment (rkuo-danswer) [reply backend/danswer/server/manage/users.py:174]: Before the code would have allowed anything into the database, so this is a strict improvement.

I don't see any issues with this on the read end of things.  At worst the email is in the database in more than one form, but we'll still only pass emails that match.

Moving forward we could add some code to renormalize existing emails, but I think we should get this change out first.

**Pull request merged**